src/core/trtserver.cc

```diff
@@ -1594,6 +1594,38 @@ TRTSERVER_ServerModelRepositoryIndex(
   return nullptr;  // success
 }
 
+// Overload to remove GRPC V2 dependency
+TRTSERVER_Error*
+TRTSERVER_ServerModelRepositoryIndexHTTP(
+    TRTSERVER_Server* server, const char*** models, uint64_t* models_count)
+{
+  ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
+
+#ifdef TRTIS_ENABLE_STATS
+  ni::ServerStatTimerScoped timer(
+      lserver->StatusManager(), ni::ServerStatTimerScoped::Kind::REPOSITORY);
+#endif  // TRTIS_ENABLE_STATS
+
+  ni::ModelRepositoryIndex model_repository_index;
+  RETURN_IF_STATUS_ERROR(
+      lserver->GetModelRepositoryIndex(&model_repository_index));
+
+  std::vector<const char*> indices;
+  for (const auto& model : model_repository_index.models()) {
+    indices.push_back(model.name().c_str());
```

Comment:
You are capturing a string here but you don't know or control the lifecycle of 'model'. You need to have a TrtModelndex class (which you expose as TRTSERVER2_ModelIndex opaque pointer) and in that capture your own copy of the strings so you can control lifecycle.